Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,270 @@ public static class BalanceStats
+            }
+            var highestBalanceEver = GetHighestBalanceEver(peopleAndBalances);
+
+            var peopleNamesWithHighestBalance = GetPeopleNamesWithSameBalance(peopleAndBalances, highestBalanceEver, "highest");
+            var formatedPeopleNames = GetFormatedPeople(peopleNamesWithHighestBalance);
+
+            // this passes the tests, but it returns highest balance not current balance as requested
+            // if it returns current balance one of the tests fails. 
+            return $"{formatedPeopleNames} had the most money ever. ¤{highestBalanceEver}.";
+        }
+
+        public static string GetFormatedPeople(string[] peopleAndBalances)
+        {
+            switch (peopleAndBalances.Length)
+            {
+                case 1:
+                    return peopleAndBalances[0];
+                case 2:
+                    return $"{peopleAndBalances[0]} and {peopleAndBalances[1]}";
+                default:
+                    // If need to this can be implemented to loop through the array, in any case only 3 different cases needed.
+                    return $"{peopleAndBalances[0]}, {peopleAndBalances[1]} and {peopleAndBalances[2]}";
+            }
+        }
+
+        public static string[] GetPeopleNamesWithSameBalance(string[] peopleAndBalances, float balance, string currentOrHighest)
+        {
+            var individualPersonAndBalance = 0f;
+            var sb = new StringBuilder();
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var person = ConvertToArray(peopleAndBalances[i]);
+
+                if (currentOrHighest.ToLowerInvariant().Equals("current"))
+                {
+                    individualPersonAndBalance = GetCurrentBalancefor(peopleAndBalances[i]);
+                }
+                if (currentOrHighest.ToLowerInvariant().Equals("highest"))
+                {
+                    individualPersonAndBalance = GetHighestBalanceFor(peopleAndBalances[i]);
+                }
+                if (individualPersonAndBalance.Equals(balance))
+                {
+                    sb.Append($"{person[0]}, ");
+                }
+            }
+            // To remove comma at the end.
+            var peopleWithSameBalance = sb.ToString().Remove(sb.ToString().Length - 2);
+            return ConvertToArray(peopleWithSameBalance);
+        }
+
+        public static float GetHighestBalanceEver(string[] peopleAndBalances)
+        {
+            var highestBalance = GetHighestBalanceFor(peopleAndBalances[0]);
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                if (highestBalance < GetHighestBalanceFor(peopleAndBalances[i]))
+                {
+                    highestBalance = GetHighestBalanceFor(peopleAndBalances[i]);
+                }
+            }
+            return highestBalance;
+        }
+        public static float GetHighestBalanceFor(string personAndBalance)
```

Comment:
"For" suffix is probably not needed. Though in some cases such style works. It's a matter of taste, I personally wouldn't write it.